pilot/pkg/model/service.go

```diff
@@ -653,3 +654,38 @@ func (s *Service) GetServiceAddressForProxy(node *Proxy) string {
 	}
 	return s.Address
 }
+
+// DeepCopy creates a clone of Service.
+// TODO : See if there is any efficient alternative to this function - copystructure can not used as is because
+// Service has sync.RWMutex that can not be copied. One alternative is to convert that to a pointer but that needs
+// lot of changes.
```

Comment:
we dont need to. the sync mutex is used by multi-cluster code. I think it should be pushed out @hzxuzhonghu

Reply:
If it is pushed out of Service, we can directly use `copystructure`